File: ruby/nth-prime/nth_prime.rb
class Prime
  def self.nth(n)
    raise ArgumentError if n < 1

    primes = [2, 3]
    current = primes.last

    while n > primes.length
      current += 2
      primes << current if is_prime?(current)
    end

    primes[n - 1]
  end

  private
    def self.is_prime?(x)
      (3..Math.sqrt(x)).step(2).all? do |y|
        x % y != 0
      end
    end
end

module BookKeeping
  VERSION = 1
end

File: javascript/nth-prime/nth-prime.js
export const prime = (i) => {
  if (i < 1) 
    throw new Error('there is no zeroth prime');

  let list = [];
  for (let a = 2; list.length < i; a += (a == 2 ? 1 : 2)) {
    if (isPrime(a))
      list.push(a);
  }
  return list[i - 1];
};

const isPrime = (a) => {
  for (let i = 3; i <= Math.sqrt(a); i += 2) {
    if (a % i === 0)
      return false;
  }
  return true;
}
